source/slang/slang-parser.cpp

```diff
@@ -2665,6 +2669,20 @@ namespace Slang
 
         AddModifiers(decl, modifiers.first);
 
+        parser->PushScope(decl);
+
+        // A `set` declaration should support declaring an explicit
+        // name for the parameter representing the new value.
+        //
+        // We handle this by supporting an arbitrary parameter list
+        // on any accessor, and then assume that semantic checking
+        // will diagnose any cases that aren't allowed.
+        //
+        if(parser->tokenReader.peekTokenType() == TokenType::LParent)
```

Comment:
Should it only support the modern param list style?

Reply:
This seems strongly related to the other comment, insofar as we need to make some decisions about Slang syntax going forward.

In general, for a given syntactic form we need to know up front whether it will use "modern" or "legacy" style. Right now the rules are that if you start a declaration with a keyword that indicates "modern" form, then you are locked into modern form for the rest of that declaration (this applies to `let`, `var`, and `func` today, and would apply to `property` and `set` with these changes)*.

For the particular case this is enabling, the detail is almost moot. The only thing we want is for somebody to be able to write:

```
set(newValue) { _someBackingField = newValue; }
```

We only really want them to specify a name for the parameter, not a type. It turns out that if you use a bare identifier in a "legacy" parameter list, then we have to interpret that as an unnamed parameter, like:

```
void foo(int); // this is a function with a single unnamed parameter of type `int`
```

whereas in a "modern" parameter list, a bare identifier is a parameter name that you forgot to give a type to:

```
func bar(x); // this is a function with a single named parameter `x`, which doesn't have a type (and thus will be an error)
```

For this specific case, interpreting things as the "modern" case gives us the right behavior for a single identifier, while still also letting the parser consume extra parameters and/or an explicit type in case we ever want to expand the scope of what is allowed.

If we did decide to use "legacy" parameter list syntax here then it would be required for the user to give the "new value" parameter an explicit type, and then we would definitely need to check that the type they gave it matches the type of the outer property.